src/sbf.cpp

```diff
@@ -0,0 +1,579 @@
+		char msg[64];
+		snprintf(msg, sizeof(msg), SBF_CONFIG_BAUDRATE, baudrate);
+
+		if (!sendMessage(msg)) {
+			continue;
+		}
+
+		if (SBF_TX_CFG_PRT_BAUDRATE != baudrate) {
+			setBaudrate(SBF_TX_CFG_PRT_BAUDRATE);
+			baudrate = SBF_TX_CFG_PRT_BAUDRATE;
+		}
+
+		if (!sendMessageAndWaitForAck(SBF_CONFIG_RESET, SBF_CONFIG_TIMEOUT)) {
+			continue;
+		}
+
+		// at this point we have correct baudrate on both ends
+
+		const char *config_cmds;
+
+		if (_output_mode == OutputMode::RTCM) {
+			config_cmds = SBF_CONFIG_RTCM;
+
+		} else {
+
+			if (_dynamic_model < 6) {
+				snprintf(msg, sizeof(msg), SBF_CONFIG_RECEIVER_DYNAMICS, "low");
+
+			} else if (_dynamic_model < 7) {
+				snprintf(msg, sizeof(msg), SBF_CONFIG_RECEIVER_DYNAMICS, "moderate");
+
+			} else if (_dynamic_model < 8) {
+				snprintf(msg, sizeof(msg), SBF_CONFIG_RECEIVER_DYNAMICS, "high");
+
+			} else {
+				snprintf(msg, sizeof(msg), SBF_CONFIG_RECEIVER_DYNAMICS, "max");
+			}
+
+			sendMessageAndWaitForAck(msg, SBF_CONFIG_TIMEOUT);
+
+			config_cmds = SBF_CONFIG;
+		}
+
+		uint8_t i = 0;
+		msg[0] = 0;
+
+		while (*config_cmds != 0) {
+			msg[i] = *config_cmds;
+
+			if (msg[i++] == '\n') {
+				msg[i] = 0;
+
+				sendMessageAndWaitForAck(msg, SBF_CONFIG_TIMEOUT);
+
+				i = 0;
+				msg[0] = 0;
+			}
+
+			config_cmds++;
+		}
+
+		if (_output_mode == OutputMode::RTCM) {
+			if (_base_settings.type == BaseSettingsType::fixed_position) {
+				snprintf(msg, sizeof(msg), SBF_CONFIG_RTCM_STATIC_COORDINATES,
+					 _base_settings.settings.fixed_position.latitude,
+					 _base_settings.settings.fixed_position.longitude,
+					 static_cast<double>(_base_settings.settings.fixed_position.altitude));
+				sendMessageAndWaitForAck(msg, SBF_CONFIG_TIMEOUT);
+				snprintf(msg, sizeof(msg), SBF_CONFIG_RTCM_STATIC_OFFSET, 0.0, 0.0, 0.0);
+				sendMessageAndWaitForAck(msg, SBF_CONFIG_TIMEOUT);
+				sendMessageAndWaitForAck(SBF_CONFIG_RTCM_STATIC1, SBF_CONFIG_TIMEOUT);
+				sendMessageAndWaitForAck(SBF_CONFIG_RTCM_STATIC2, SBF_CONFIG_TIMEOUT);
+			}
+		}
+
+		break;
+	}
+
+	if (baud_i >= sizeof(baudrates) / sizeof(baudrates[0])) {
+		return -1; // connection and/or baudrate detection failed
+	}
+
+	_configured = true;
+	return 0;
+}
+
+bool
+GPSDriverSBF::sendMessage(const char *msg)
+{
+	SBF_DEBUG("Send MSG: %s", msg);
+
+	// Send message
+	int length = static_cast<int>(strlen(msg));
+
+	if (write(msg, length) != length) {
+		return false;
+	}
+
+	return true;
+}
+
+bool
+GPSDriverSBF::sendMessageAndWaitForAck(const char *msg, const int timeout)
+{
+	SBF_DEBUG("Send MSG: %s", msg);
+
+	// Send message
+	int length = static_cast<int>(strlen(msg));
+
+	if (write(msg, length) != length) {
+		return false;
+	}
+
+	// Wait for acknowledge
+	// For all valid set -, get - and exe -commands, the first line of the reply is an exact copy
+	// of the command as entered by the user, preceded with "$R:"
+	char buf[GPS_READ_BUFFER_SIZE];
+	size_t offset = 0;
+	memset(buf, 0, sizeof(buf));
```

Comment:
No need to zero-initialize it, just make sure buf is null-terminated before calling `strstr`.

Reply:
Implemented.